pkg/apis/duck/v1alpha1/pubsub_lifecycle.go

```diff
@@ -20,23 +20,44 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-// MarkTopicNotReady sets the condition that the PubSub Topic is not ready and why.
-func (s *PubSubStatus) MarkTopicNotReady(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
+// MarkTopicFalse sets the condition that the PubSub Topic is not ready and why.
+func (s *PubSubStatus) MarkTopicFalse(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
```

Comment:
At first, we only need to differentiate between "ready" and "notReady". Now we need to differentiate among "True", "False" and "Unknown". I just followed the naming convention of `/knative.dev/pkg/apis/condition_set.go`

Reply:
I think the naming in `knative.dev/pkg/apis/condition_set.go` makes sense because the condition type is in the function parameters. It's pretty natural to say "mark condition type t to false". But it's not as quite natural to say "mark topic to false" because what you really do is "mark the topic ready condition to false".

An alternative is to do what eventing/serving does. E.g.

Use `MarkTopicReady` for ready=true, `MarkTopicFailed` for ready=false, `MarkTopicUnknown` for ready=unknown. Up to you.

Reply:
In this discussion thread, https://github.com/knative/eventing/pull/2341#discussion_r361755665
. It seems like "Failed" is only for batch type resources and "False" is for living type resources. But currently in eventing/serving, it does not follow this standard.

Reply:
Interesting... I don't feel strongly about this. I simply prefer `MarkTopicNotReady` to `MarkTopicFalse`. I'm also fine with `MarkTopicFailed`. Will leave it to you to decide.

Reply:
hmm, I agree with @yolocs, I'd rather have `MarkTopicNotReady` instead of `MarkTopicFalse`... Similar with True conditions... `MarkTopicReady` instead of `MarkTopicTrue`... and I guess `MarkTopicUnknown` for the remaining sounds good...

Reply:
I prefer `MarkTopicFailed` over `MarkTopicNotReady` since it's more clear. Is that ok?